File: src/www/api/routes/deposit.php
<?php

// ==============================================================================
//
// This file is part of the WelStory.
//
// Create by Welfony Support <support@welfony.com>
// Copyright (c) 2012-2014 welfony.com
//
// For the full copyright and license information, please view the LICENSE
// file that was distributed with this source code.
//
// ==============================================================================

use Welfony\Controller\API\DepositController;

$app->get('/users/:userId/deposit', function ($userId) use ($app) {
    $ctrl = new DepositController();
    $ctrl->listByUser($userId);
})->conditions(array('userId' => '\d{1,10}'));

$app->post('/users/:userId/deposit', function ($userId) use ($app) {
    $ctrl = new DepositController();
    $ctrl->create($userId);
})->conditions(array('userId' => '\d{1,10}'));

File: src/ios/App/Models/Deposit.m
// ==============================================================================
//
// This file is part of the WelHair
//
// Create by Welfony <support@welfony.com>
// Copyright (c) 2013-2014 welfony.com
//
// For the full copyright and license information, please view the LICENSE
// file that was distributed with this source code.
//
// ==============================================================================

#import "Deposit.h"

@implementation Deposit

- (Deposit *)initWithDic:(NSDictionary *)dictionary
{
    self = [super init];
    if (self) {
        self.id = [[dictionary objectForKey:@"DepositId"] intValue];
        self.depositNo = [dictionary objectForKey:@"DepositNo"];
        self.amount = [[dictionary objectForKey:@"Amount"] floatValue];
        self.status = [[dictionary objectForKey:@"Status"] intValue];

        self.createdDate = [[NSDate dateFormatter] dateFromString:[dictionary objectForKey:@"CreateTime"]];
    }

    return self;
}

@end
